File: q1/binary_search.js
function binary_search (L, k) {
    console.log(L, k, mid);

    if (L.length > 1) {
        if (L[mid] > k) {
            return binary_search(L.slice(0, mid), k);
        } else if (L[mid] < k) {
            return binary_search(L.slice(mid, L.length), k);
        }
    }

    return (L[mid] == k)
}

File: q1/binary_search.py
def binary_search(L, k):
    mid = len(L)/2

    if len(L) > 1:
        if L[mid] < k:
            return binary_search(L[mid:], k)
        elif L[mid] > k:
            return binary_search(L[:mid], k)
    
    return L[mid] == k
